consensus/service/src/api/client_api_service.rs

```diff
@@ -1694,4 +1694,91 @@ mod client_api_tests {
 
         assert!(submitted_values.lock().unwrap().is_empty());
     }
+
+    // Very simple test of the session-tracking feature
+    #[test_with_logger]
+    #[serial(counters)]
+    fn test_session_tracking_gets_populated(logger: Logger) {
+        let mut consensus_enclave = MockConsensusEnclave::new();
+        {
+            // Return a TxContext that contains some KeyImages.
+            let tx_context = TxContext {
+                key_images: vec![KeyImage::default(), KeyImage::default()],
+                ..Default::default()
+            };
+
+            consensus_enclave
+                .expect_client_tx_propose()
+                .times(1)
+                .return_const(Ok(tx_context));
+        }
+
+        let scp_client_value_sender = Arc::new(
+            |_value: ConsensusValue,
+             _node_id: Option<&NodeID>,
+             _responder_id: Option<&ResponderId>| {
+                // TODO: store inputs for inspection.
+            },
+        );
+
+        let num_blocks = 5;
```

Comment:
nit: Make a `
```suggestion
        const NUM_BLOCKS: u64= 5;
```

Not sure what type it should be, just figured it was `u64`.